vortex-layout/src/scan/mod.rs

```diff
@@ -211,6 +210,45 @@ impl<D: ScanDriver> ScanBuilder<D> {
     }
 }
 
+/// Compute masks of field paths referenced by the projection and filter in the scan.
+///
+/// Projection and filter must be pre-simplified.
+fn filter_and_projection_masks(
```

Comment:
This function kind of does two things interleaved? Can you just have a single function that returns a FieldMask from an ExprRef + DType?